ww3tools/ProcSat_interpolation.py

```diff
@@ -0,0 +1,515 @@
+
+    def extract_reference_times(file_path):
+        ds = xr.open_dataset(file_path)
+        reference_time = ds.time.attrs['reference_time']
+        reference_date = ds.time.attrs['reference_date']
+        ds.close()  # Close the dataset after extraction
+        return reference_time, reference_date
+
+    model_reference_time_unix = extract_reference_times(model_data_files[0])[0]  #added
+    reference_time_unix = extract_reference_times(model_data_files[0])[0]  # Returns Unix timestamp as integer or float
+
+    # Load and concatenate the model data files
+    model_data_list = [xr.open_dataset(file) for file in model_data_files]
+    model_data_combined = xr.concat(model_data_list, dim='time')
+    model_data_combined.close()  # Close the files after loading
+
+    # Convert longitudes from -180 to 180 to 0 to 360, if necessary
+    model_longitudes = model_data_combined['longitude'].values
+    if model_longitudes.min() < 0:
+        model_longitudes[model_longitudes < 0] += 360
+        model_data_combined = model_data_combined.assign_coords(longitude=model_longitudes)
+
+    # Convert model time to UNIX time and ensure it's double
+    model_time_seconds = np.array((model_data_combined['time'].values - np.datetime64('1970-01-01T00:00:00')) / np.timedelta64(1, 's')).astype('double')
+    max_model_time = model_time_seconds.max()
+
+    # Defining model_times correctly for use in interpolation
+    model_times = model_time_seconds
+
+    # Load satellite data
+    satellite_data = xr.open_dataset(satellite_file)
+    satellite_longitudes = satellite_data['longitude'].values
+    satellite_longitudes[satellite_longitudes < 0] += 360
+    satellite_times_unix = np.array((satellite_data['time'].values - np.datetime64('1970-01-01T00:00:00')) / np.timedelta64(1, 's')).astype('double')
+    filtered_satellite_times_unix = satellite_times_unix[satellite_times_unix <= max_model_time]
+    filtered_satellite_latitudes = satellite_data['latitude'].values[:len(filtered_satellite_times_unix)]
+    filtered_satellite_longitudes = satellite_longitudes[:len(filtered_satellite_times_unix)]
+
+    # Attempt to read the satellite name from the attributes or a variable
+    if 'sat_name' in satellite_data.variables:
+        satellite_name = satellite_data['sat_name'].values[0]  # Assuming it's stored as a variable
+    elif 'satellite_name' in satellite_data.attrs:
+        satellite_name = satellite_data.attrs['satellite_name']  # Assuming it's stored as an attribute
+    else:
+        satellite_name = "UnknownSatellite"  # Default name if not found
+
+    # Define the Model's Spatial Bounds
+    min_latitude = model_data_combined['latitude'].min()
+    max_latitude = model_data_combined['latitude'].max()
+    min_longitude = model_data_combined['longitude'].min()
+    max_longitude = model_data_combined['longitude'].max()
+
+    # Function to create and apply a spatial mask
+    def spatial_mask(satellite_latitudes, satellite_longitudes):
```

Comment:
I noticed that the grib2 routine takes into account model values, is there a reason that we do this for grib2 and not netcdf?

Reply:
Issue addressed.